cmd/witr/main.go

```diff
@@ -2,263 +2,211 @@ package main
-
-	// Reorder os.Args so all flags (with their values) come before positional arguments
-	reordered := []string{os.Args[0]}
-	var positionals []string
-	i := 1
-	for i < len(os.Args) {
-		arg := os.Args[i]
-		if len(arg) > 0 && arg[0] == '-' {
-			reordered = append(reordered, arg)
-			// If this flag takes a value (not a bool flag), keep the value with it
-			if flagNeedsValue(arg) && i+1 < len(os.Args) && os.Args[i+1][0] != '-' {
-				reordered = append(reordered, os.Args[i+1])
-				i++
-			}
-		} else {
-			positionals = append(positionals, arg)
-		}
-		i++
-	}
-	reordered = append(reordered, positionals...)
-	os.Args = reordered
-
-	pidFlag := flag.String("pid", "", "pid to explain")
-	portFlag := flag.String("port", "", "port to explain")
-	shortFlag := flag.Bool("short", false, "short output")
-	treeFlag := flag.Bool("tree", false, "tree output")
-	jsonFlag := flag.Bool("json", false, "output as JSON")
-	warnFlag := flag.Bool("warnings", false, "show only warnings")
-	noColorFlag := flag.Bool("no-color", false, "disable colorized output")
-	envFlag := flag.Bool("env", false, "show only environment variables for the process")
-	helpFlag := flag.Bool("help", false, "show help")
-
-	flag.Parse()
-
-	if *versionFlag {
-		fmt.Printf("witr %s (commit %s, built %s)\n", version, commit, buildDate)
-		os.Exit(0)
-	}
 	// To embed version, commit, and build date, use:
 	// go build -ldflags "-X main.version=v0.1.0 -X main.commit=$(git rev-parse --short HEAD) -X 'main.buildDate=$(date +%Y-%m-%d)'" -o witr ./cmd/witr
-	if *envFlag {
-		var t model.Target
-		switch {
-		case *pidFlag != "":
-			t = model.Target{Type: model.TargetPID, Value: *pidFlag}
-		case *portFlag != "":
-			t = model.Target{Type: model.TargetPort, Value: *portFlag}
-		case len(flag.Args()) > 0:
-			t = model.Target{Type: model.TargetName, Value: flag.Args()[0]}
-		default:
-			printHelp()
-			os.Exit(1)
-		}
 
-		pids, err := target.Resolve(t)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
-			os.Exit(1)
-		}
-		if len(pids) > 1 {
-			fmt.Print("Multiple matching processes found:\n\n")
-			for i, pid := range pids {
-				cmdline := procpkg.GetCmdline(pid)
-				fmt.Printf("[%d] PID %d   %s\n", i+1, pid, cmdline)
-			}
-			fmt.Println("\nRe-run with:")
-			fmt.Println("  witr --pid <pid> --env")
-			os.Exit(1)
-		}
-		pid := pids[0]
-		procInfo, err := procpkg.ReadProcess(pid)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
-			os.Exit(1)
-		}
-		if *jsonFlag {
-			type envOut struct {
-				Command string   `json:"Command"`
-				Env     []string `json:"Env"`
-			}
-			out := envOut{Command: procInfo.Cmdline, Env: procInfo.Env}
-			enc, _ := json.MarshalIndent(out, "", "  ")
-			fmt.Println(string(enc))
-		} else {
-			output.RenderEnvOnly(procInfo, !*noColorFlag)
-		}
-		return
-	}
+	rootCmd := &cobra.Command{
+		Use:   "witr [--pid N | --port N | name] [--short] [--tree] [--json] [--warnings] [--no-color] [--env] [--help]",
```

Comment:
you could consider making the usage string more compact eg.:
```suggestion
		Use:   "witr [flags...]",
```

Everything else lgtm

Reply:
Hmm I'm actually not sure how this should be done. I see now Cobra adds "[flags]" to the end, and as you've written it outputs as before, but with:

```
Usage:
	witr [flags...] [flags]
```
This can be overriden with `SetUsageTemplate` but it seems that overrides the entire help message. We can write a longer template, e.g.:

```
{{.Long}}

Usage:
	witr [flags...]

Flags:
{{.LocalFlags.FlagUsages}}
```
which outputs like as previously with the improved usage message at the risk of it being a little more brittle? This is the working change in 398b044.